js/compiler_test.ts

```diff
@@ -249,120 +250,105 @@ test(function compilerInstance() {
-
-  const lastGlobalEval = globalEvalStack.pop();
-  assertEqual(lastGlobalEval, fooBarTsOutput);
-  const lastCodeFetch = codeFetchStack.pop();
-  assertEqual(lastCodeFetch, {
-    moduleSpecifier: "foo/bar.ts",
-    containingFile: "/root/project"
-  });
-  const lastCodeCache = codeCacheStack.pop();
-  assertEqual(lastCodeCache, {
-    fileName: "/root/project/foo/bar.ts",
-    sourceCode: fooBarTsSource,
-    outputCode: fooBarTsOutput
-  });
+test(function compilerResolveModule() {
+  setup();
+  const moduleMetaData = compilerInstance.resolveModule(
+    "foo/baz.ts",
+    "/root/project"
+  );
+  assertEqual(moduleMetaData.sourceCode, fooBazTsSource);
+  assertEqual(moduleMetaData.outputCode, fooBazTsOutput);
+  assert(!moduleMetaData.hasRun);
+  assert(!moduleMetaData.deps);
+  assertEqual(moduleMetaData.exports, {});
+  assertEqual(moduleMetaData.scriptVersion, "1");
+
+  assertEqual(codeFetchStack.length, 1, "Only initial module is resolved.");
+  teardown();
 });
 
-test(function compilerRunMultiModule() {
-  // equal to `deno foo/baz.ts`
-  reset();
-  let factoryRun = false;
-  mockDeps = ["require", "exports", "compiler"];
-  mockFactory = (...deps: any[]) => {
-    const [_require, _exports, _compiler] = deps;
-    assertEqual(typeof _require, "function");
-    assertEqual(typeof _exports, "object");
-    assertEqual(_compiler, compiler);
-    factoryRun = true;
-    Object.defineProperty(_exports, "__esModule", { value: true });
-    _exports.foo = "bar";
-    // it is too complicated to test the outer factory, because the localised
-    // make define already has a reference to this factory and it can't really
-    // be easily unwound.  So we will do what we can with the inner one and
-    // then just clear it...
-    mockDeps = undefined;
-    mockFactory = undefined;
+test(function compilerGetModuleDependencies() {
+  setup();
+  const bazDeps = ["require", "exports", "./bar.ts"];
+  const bazFactory = () => {
+    throw new Error("Unexpected factory call");
   };
-
-  const result = compilerInstance.run("foo/baz.ts", "/root/project");
-  assert(result instanceof ModuleMetaData);
-  // we have mocked that foo/bar.ts is already cached, so two fetches,
-  // but only a single cache
-  assertEqual(codeFetchStack.length, 2);
-  assertEqual(codeCacheStack.length, 1);
-  // because of the challenges with the way the module factories are generated
-  // we only get one invocation of the `globalEval` mock.
-  assertEqual(globalEvalStack.length, 1);
-  assert(factoryRun);
+  const barDeps = ["require", "exports", "compiler"];
+  const barFactory = () => {
+    throw new Error("Unexpected factory call");
+  };
+  mockDepsStack.push(barDeps);
+  mockFactoryStack.push(barFactory);
+  mockDepsStack.push(bazDeps);
+  mockFactoryStack.push(bazFactory);
+  const deps = compilerInstance.getModuleDependencies(
+    "foo/baz.ts",
+    "/root/project"
+  );
+  assertEqual(deps, ["/root/project/foo/bar.ts", "/root/project/foo/baz.ts"]);
```

Comment:
Awesome!!!

Reply:
The only improvement, which we would have to change the privileged is that this will resolve to the cached name instead of the source file name. I suspect it would be handy to know both in the future.

Reply:
I was toying with the idea of keeping that information private to the privlaged side. An implementation detail. But I would be open to exposing it if there’s a good reason to.